src/Sylius/Component/Archetype/Builder/ArchetypeBuilderInterface.php

```diff
@@ -0,0 +1,29 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sylius\Component\Archetype\Builder;
+
+use Sylius\Component\Archetype\Model\ArchetypeSubjectInterface;
+use Sylius\Component\Archetype\Model\ArchetypeInterface;
+
+/**
+ * @author Paweł Jędrzejewski <pawel@sylius.org>
+ * @author Adam Elsodaney <adam.elso@gmail.com>
+ */
+interface ArchetypeBuilderInterface
+{
+    /**
+     * Build the archetype of product.
+     *
+     * @param ArchetypeSubjectInterface $subject
+     */
+    public function build(ArchetypeSubjectInterface $subject);
```

Comment:
Should `ArchetypeSubjectInterface` perhaps not extend the `AttributeSubjectInterface` and `VariableInterface` ? I'm thinking a scenario where an archetype may only use attributes, but not options, yet the `ArchetypeSubjectInterface` requires the implementation of method to manipulate options.

Instead, the building of attributes or options can be taking care in an implementation of `ArchetypeBuilderInterface::build()`.